### PR TITLE
Support merging primitive lists

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

**Comment by pwittrock** [comment]
cc @monopole

File: kyaml/yaml/walk/associative_sequence.go [hunk 1]
```diff
@@ -19,21 +19,69 @@ func (l *Walker) walkAssociativeSequence() (*yaml.RNode, error) {
...
+	if strategy == "" && key == "" { // neither strategy nor not present in the schema -- infer the key
 		// find the list of elements we need to recursively walk
 		key, err = l.elementKey()
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	values := l.elementValues(key)
+	// non-primitive associative list -- merge the elements
```
Comment (monopole) [inline kyaml/yaml/walk/associative_sequence.go:34]: can you drop a little more context here on supported values for key and strategy?

File: kyaml/yaml/walk/associative_sequence.go [hunk 2]
```diff
@@ -19,21 +19,69 @@ func (l *Walker) walkAssociativeSequence() (*yaml.RNode, error) {
...
+	if key != "" {
```
Comment (monopole) [inline kyaml/yaml/walk/associative_sequence.go:35]: throw 'if' body into a method with aforementioned context??

**Pull request merged**